cmd/setup-etcd-environment/run.go

```diff
@@ -193,70 +249,79 @@ func setExportEnv(etcdName, dns string) (map[string]string, error) {
-	for _, member := range hostEtcdEndpoint.Subsets[0].Addresses {
-		if member.Hostname == "etcd-bootstrap" {
-			endpoints = append(endpoints, "https://etcd-bootstrap."+runOpts.discoverySRV+":2379")
-			break
-		}
-	}
-	if len(ep.Subsets) != 1 {
-		return nil, fmt.Errorf("openshift-etcd/etcd endpoint subset length should be %d, found %d", 1, len(ep.Subsets))
-	}
-	for _, s := range ep.Subsets[0].Addresses {
-		endpoints = append(endpoints, "https://"+s.IP+":2379")
-	}
-
-	exportEnv["ENDPOINTS"] = strings.Join(endpoints, ",")
-	return exportEnv, nil
+	return memberList
 }
 
-// setBootstrapEnv populates and returns a map based on envs from file.
-func setBootstrapEnv(envFile, discoverySRV string, bootstrapSRV bool) (map[string]string, error) {
+// getBootstrapEnv populates ENV variables used for bootstrapping etcd with SRV and recovery. We take care
+// to persist important ENV during reboot. We also make sure to not mix ENV that might cause etcd to fail to start.
+func (s *SetupEnv) getBootstrapEnv() (map[string]string, error) {
 	bootstrapEnv := make(map[string]string)
-	if _, err := os.Stat(envFile); !os.IsNotExist(err) {
-		recoveryEnv, err := godotenv.Read(envFile)
+	if _, err := os.Stat(s.opts.outputFile); !os.IsNotExist(err) {
+		// read existing ENV file
+		existingEnv, err := godotenv.Read(s.opts.outputFile)
 		if err != nil {
 			return nil, err
 		}
-		// persist any observed envs used for recovery
+		// persist any observed envs used for recovery. We want to make sure that ETCD_INITIAL_CLUSTER and
+		// ETCD_INITIAL_CLUSTER_STATE survive reboot.
 		for _, val := range []string{"INITIAL_CLUSTER", "INITIAL_CLUSTER_STATE"} {
 			rkey := fmt.Sprintf("ETCD_%s", val)
-			if recoveryEnv[rkey] != "" {
-				bootstrapEnv[val] = recoveryEnv[rkey]
+			if existingEnv[rkey] != "" {
+				bootstrapEnv[val] = existingEnv[rkey]
 			}
 		}
 	}
-	// define srv discovery if enabeled and not in recovery mode
-	if len(bootstrapEnv) == 0 && bootstrapSRV {
-		bootstrapEnv["DISCOVERY_SRV"] = discoverySRV
+
+	// Define SRV discovery variables if enabeled and not in recovery mode. This is important because we can
+	// not mix `ETCD_DISCOVERY_SRV` with ETCD_INITIAL_CLUSTER` or etcd will fail to start.
+	if len(bootstrapEnv) == 0 && s.opts.bootstrapSRV {
+		bootstrapEnv["DISCOVERY_SRV"] = s.opts.discoverySRV
 	}
 	return bootstrapEnv, nil
 }
 
-func ipAddrs() ([]string, error) {
+// isEtcdScaling performs a series of checks to see if etcd is currently attempting to scale from the
+// `cluster-etcd-operator` and returns true if yes. If false we can conclude that the member is already
+// part of the cluster or we have/are bootstrapping etcd using SRV discovery.
+func (s *SetupEnv) isEtcdScaling() bool {
+	if _, err := os.Stat(fmt.Sprintf("%s/member", s.etcdDataDir)); os.IsNotExist(err) && !s.opts.bootstrapSRV && inCluster() {
+		return true
+	}
+	return false
+}
+
+func (s *SetupEnv) setClient() error {
+	if inCluster() && !s.opts.bootstrapSRV {
```

Comment:
Not sure about this but can we

s/inCluster() && !s.opts.bootstrapSRV/inCluster() 

and push the check `!s.opts.bootstrapSRV/inCluster()` in function `inCluster`?

Reply:
we can consider this as TODO